### ISSUE TITLE
aws-cdk-lib.aws_rds module DatabaseInstance: Support the creation of MSSQL DB bound to self managed AD instance.

### ISSUE THREAD

**Issue by zcvetkovicgscf** [aws/aws-cdk#28036]
### Describe the feature

As recently announced AWS-RDS MsSQL Instances support  the integration with self managed AD Instances, look at: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_SQLServer_SelfManagedActiveDirectory.html#USER_SQLServer_SelfManagedActiveDirectory.Managing

It Seams that cdk_rds is not supporting this feature.

### Use Case

Create a RDS-MSSQL Instance coupled to a self managed Active Directory. 

### Proposed Solution

extend the DatabaseInstanceProps by needed parameters e.g.

domain-fqdn
domain-dns-is
domain-ou
domain-auth-secret-arn

an implement it DatabaseInstance constructor alt least for MSSQL engine.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.99.1

### Environment details (OS name and version, etc.)

linux, macosx and GitHub

**Comment by pahud** [comment]
I am not 100% sure if cloudformation has this support but yes, we could support that if it does.

**Comment by zcvetkovicgscf** [comment]
Thanks @pahud for the quick triage:
have a look - the given parameters are already defined for CloudFormation Resources 
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.CfnDBInstance.html:

  domain: 'domain',
  domainAuthSecretArn: 'domainAuthSecretArn',
  domainDnsIps: ['domainDnsIps'],
  domainFqdn: 'domainFqdn',
  domainIamRoleName: 'domainIamRoleName',
  domainOu: 'domainOu',

so I guess this it what you meant when looking if cloudformation has this support?

**Comment by pahud** [comment]
Yes, sounds like it could be a new L2 for that. Please help us prioritize with 👍 .
